### PR TITLE
[messagebus, bridge] Remove messagebus cross-cluster dependency

### LINKED ISSUE #7468: Remove messagebus shovel
Remove shovel by having ws-manager-bridge "pull" state from each cluster
Goal: Replace messagebus cross-clusters dependency with the (already existing) workspace-cluster dependency.

### PR DESCRIPTION
**Pull request by geropl**
## Description
This PR contains two complementary changes:
 1. remove the messagebus shovel config, and all non-local exchanges (`gitpod.ws`) and use local exchanges instead (`gitpod.ws.local`)
 2. change `ws-manager-bridge` so that bridges that are non-`govern`ing:
   a. still don't write updates to the DB,
   b. but do the same derivation of updates,
   c. and distributes those over their cluster's local messagebus

**Note**: This PR can be reviewed commit by commit!

## Pain points
[This polling and hashing](https://github.com/gitpod-io/gitpod/pull/7523/commits/3eacd3388e19bd3197a2291e278cd8ae4f85ed1b#diff-dda1d30e0a84afa9ad6c941cf99ebef2211138c7287c659b041cd05bbbcaebb7R41-R48) feel quite expensive, especially in scenarios where things go haywire already (a lot of workspaces hanging in preparing). But I see no way around it until we pushed image-builder into workspace clusters (https://github.com/gitpod-io/gitpod/issues/7845). :shrug:  Once that's done we should be able to just delete PreparingUpdateEmulator :wastebasket: .

## Deployment
This PR is meant to be a drop-in replacement for the current setup, which means:

    behavior before and after should be identical (order of updates, state persisted to DB before update sent, except for cross-cluster stuff, etc.)
    all currently running workspaces should behave as usual, e.g. keep running smoothly
    rolling this out will require a messagebus restart, but apart from that behave like a regular deployment

## ToDo:
 - [x] ~think about how to properly test this, ideally before~ **on** staging
 - [x] create a cleanup PR for the ops repo https://github.com/gitpod-io/ops/pull/666
 - [x] solve the "imagebuild in different cluster" case
   - [x] w.r.t. instance updates: [`PreparingUpdateEmulator`](https://github.com/gitpod-io/gitpod/pull/7523/files#diff-5c377b2cad37712ee6f512a499f1069f7641af5aab7dceaa7e8d033fc23d4780R90-R93)
   - [x] ~w.r.t. image build logs~ solved in: #7899

## Related Issue(s)

Fixes #7468

### cluster-local works as before
 - start a workspace with devtools open, and watch the websocket messages that keep flowing in
 - note the onInstaceUpdate notifications
 - note how the workspace starts as expected

### cross-cluster
We'll test this on staging as the setup is already there, and an alternative setup would eat too much time at the moment.

## Release Notes

```release-note
messagebus: remove cross-cluster dependency
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **metcalfc** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @metcalfc` in a comment when ready.

Associated issue: *#7468*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-messagebus/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-messagebus/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **corneliusludmann** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @corneliusludmann` in a comment when ready.

Associated issue: *#7468*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/gitpod-messagebus/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-messagebus/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **metcalfc** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @metcalfc` in a comment when ready.

Associated issue: *#7468*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-messagebus/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-messagebus/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **mrsimonemms**, **utam0k** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms @utam0k` in a comment when ready.

Associated issue: *#7468*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-messagebus/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-messagebus/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/image-builder-api/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-api/OWNERS)**
- **[components/image-builder-mk3/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/image-builder-mk3/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **geropl** and additionally assign **mrsimonemms** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mrsimonemms` in a comment when ready.

Associated issue: *#7468*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**
- **[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)**
- **[components/gitpod-messagebus/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-messagebus/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**
- **[installer/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/OWNERS)**
- **[installer/pkg/components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by geropl** [comment]
/hold

Don't merge before tomorrows deployment :slightly_smiling_face:

**Review by JanKoehnlein — APPROVED** [review]
LGTM codewise. Let's merge and test it on staging

**Review by corneliusludmann — APPROVED** [review]

**Pull request merged**
